lib/web/apiserver.go

```diff
@@ -589,9 +589,21 @@ func NewHandler(cfg Config, opts ...HandlerOption) (*APIHandler, error) {
 				// so we only set the default CSP on that page
 				parts := strings.Split(r.URL.Path, "/")
 				// grab the FQDN from the URL to allow in the connect-src CSP
-				applicationURL := "https://" + parts[3] + ":*"
+				applicationURL := "https://" + parts[3]
```

Comment:
Do we need to check that `len(parts) >= 4` prior to reading `parts[3]` here?

Reply:
no it's not required. because we first match by prefix `/web/launch/` which is guaranteed to be of length 4. However, I added a check if `parts[3] == ""` because that can be a case.

Any paths that come after the application URL is stored as a `path` query param when the backend redirects user to web/launcher.

The only time where paths can be greater than 4 is when someone tries to construct web launcher URL themselves (which isn't something we support)